stdlib/asm/collections/smt.masm

```diff
@@ -1,14 +1,421 @@
-use.std::mem
+use.std::crypto::hashes::rpo
```

Comment:
Great implementation! Consider adding a randomized round-trip test to strengthen coverage. A test that creates non-empty SMTs with random key-value pairs, uses smt::set to insert values, then smt::get to verify round-trip correctness would help validate multi-leaf functionality in realistic scenarios beyond deterministic test cases.